File: Creational/Lab3/C#/Realtor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace C_
{
    public class Realtor : Person
    {
        double YearsOfWork{ get; set; }
        public Realtor(int id, string nameAndSurname, string phone, string email, double yearsOfWork):base(id, nameAndSurname, phone, email) {
            YearsOfWork = yearsOfWork;
        }

        public Realtor(Realtor realtor):base(realtor.Id, realtor.NameAndSurname, realtor.Phone, realtor.Email) { }

        public override string ToString()
        {
            return $"Realtor:\nId:{Id}, Name and surname: {NameAndSurname},\nPhone number: {Phone}, Email: {Email}, Years of work {YearsOfWork}";
        }
    }
}

File: Behavioral/Lab21/Typescript/src/Entities/realtor.ts
import { Person } from './person';

export class Realtor extends Person {
    public yearsOfWork: number;
    constructor(nameAndSurname: string, phone: string, email: string, yearsOfWork: number){
        super(nameAndSurname, phone, email);
        this.yearsOfWork = yearsOfWork;
    }
    public static newRealtorFromObject(realtor: Realtor): Realtor {
        let r = new Realtor(realtor.nameAndSurname, realtor.phone, realtor.email, realtor.yearsOfWork);
        return r;
    }

    public toString(): string {
        return `Realtor:\nId:${this.id}, Name and surname: ${this.nameAndSurname},\nPhone number: ${this.phone}, Email: ${this.email}, Years of work ${this.yearsOfWork}`;
    }
}